### PR TITLE
Bulk scan admin - upload and show table

### PR DESCRIPTION
**Pull request by ganeshkumar78**
changes:
1. New endpoint to upload csv file
2. validate csv file format, validate barcodes and call private api to save barcode scan data
3. Show valid barcode data in the UI(Shows the original data for now)

### CODE REVIEW

File: microsetta_admin/server.py [hunk 1]
```diff
@@ -773,6 +773,41 @@ def scan():
...
 
+@app.route('/scan_bulk', methods=['GET', 'POST'])
+def scan_bulk():
+    if request.method == 'GET':
+
+        return render_template("scan_bulk.html",
+                               **build_login_variables())
+                                   # resource=None,
+                                   # search_error=None,
+                                   # projects=projects)
```
Comment (wasade) [inline microsetta_admin/server.py:784]:
```suggestion
```

File: microsetta_admin/server.py [hunk 2]
```diff
@@ -773,6 +773,41 @@ def scan():
...
+
+def _bulk_scan_post(request):
+    # Read uploaded csv and post it to private api    
+    if 'bulk_csv_file' in request.files:
+        request_file = request.files['bulk_csv_file']
+        records = pd.read_csv(request_file, dtype=str).to_dict(orient='records')
+        if records:
+            status, result = APIRequest.post(
+                '/api/admin/bulk_scan', json=records)
+            print(result)
```
Comment (wasade) [inline microsetta_admin/server.py:806]:
```suggestion
```

File: microsetta_admin/templates/scan_bulk.html [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+            alert("A scan cannot be saved without selecting a sample_status.");
+        }
+
+        return result;
+    }
+
+    $(function(){
+        document.search_form.sample_barcode.focus();
+
+        old_status = new NamedExpression("OLD_SAMPLE_STATUS", "{{latest_status}}")
```
Comment (wasade) [inline microsetta_admin/templates/scan_bulk.html:43]: I didn't see it, but perhaps missed it, where in biocore/microsetta-private-api#496 sample status or other details about a barcode are returned?

File: microsetta_admin/templates/scan_bulk.html [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        new DelegateOutput(function(should_send){
+            if (should_send)
+                $("#scan_button").prop("value", "Add Scan And Notify User")
+            else
+                $("#scan_button").prop("value", "Add Scan")
+        }, should_send_email);
+
+        // Hide/Show email template inputs based on sample_status select
+        new DelegateOutput(function(selected_status){
+            if (selected_status == "sample-has-inconsistencies" ||
```
Comment (wasade) [inline microsetta_admin/templates/scan_bulk.html:69]: Please use `===` here and below

File: microsetta_admin/templates/scan_bulk.html [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        <table>
+            <tr>
+                <td><label for="import_csv">import .csv: </label></td>
+                <td><input type="text" name="upload_csv_textfield" id="upload_csv_textfield"></td>
+                <td><input type="file"
+                           name="bulk_csv_file"
+                           style="color:transparent;"
+                           onchange="document.getElementById('upload_csv_textfield').value = this.value.split('\\').pop().split('/').pop()"></td>
+
+<!--                <td><input type="text" name="upload_csv" id="upload_csv"></td>-->
```
Comment (wasade) [inline microsetta_admin/templates/scan_bulk.html:185]:
```suggestion
```

File: microsetta_admin/templates/scan_bulk.html [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+<script src="/static/vendor/DataTables/Buttons-1.6.2/js/dataTables.buttons.min.js"></script>
+<script src="/static/vendor/DataTables/Buttons-1.6.2/js/buttons.html5.min.js"></script>
+<script>
+    function remove_error_messages () {
+        block = document.getElementById("search_errors");
+        if (block !== null) {
+            block.innerHTML = "";
+        }
+    };
+        $(document).ready(function() {
```
Comment (wasade) [inline microsetta_admin/templates/scan_bulk.html:16]:
```suggestion
    $(document).ready(function() {
```

File: microsetta_admin/templates/scan_bulk.html [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+        {% if resource is not none %}
+            $('#search_results').DataTable({
```
Comment (wasade) [inline microsetta_admin/templates/scan_bulk.html:18]: Perhaps I missed it, but I'm not finding this `id` in the document?

File: setup.py [hunk 8]
```diff
@@ -1,49 +1,356 @@
-# ----------------------------------------------------------------------------
```
Comment (wasade) [inline setup.py:1]: Please undelete `setup.py`

File: setup.py [hunk 9]
```diff
@@ -1,49 +1,356 @@
...
-        'static/vendor/DataTables/DataTables-1.10.21/*',
-        'static/vendor/DataTables/DataTables-1.10.21/css/*',
-        'static/vendor/DataTables/DataTables-1.10.21/images/*',
-        'static/vendor/DataTables/DataTables-1.10.21/js/*',
-        'static/vendor/DataTables/FixedColumns-3.3.1/*',
-        'static/vendor/DataTables/FixedColumns-3.3.1/css/*',
-        'static/vendor/DataTables/FixedColumns-3.3.1/jss/*',
-        'authrocket.pubkey']},
-)
+{% extends "sitebase.html" %}
```
Comment (wasade) [inline setup.py:1]: These HTML changes are in a `setup.py` which is unexpected and likely incorrect

Comment (wasade) [reply setup.py:1]: At least some of what's below looks to be duplicate with what's in `scan_bulk.html`, and should definitely not be in `setup.py`. 

Please make sure to run the project and render the templates locally before pushing commits.

**Review by wasade — CHANGES_REQUESTED** [review]
Thanks, @ganeshkumar78. I did not fully evaluate this PR as there are incorrect changes to `setup.py` such that the project would not run properly. Please verify the code locally before the next set of commits are pushed.

**Comment by ganeshkumar78** [comment]
@wasade , I have accidentally raised PR's for review. There were supposed to be draft PR's. Its still work in progress. I will make sure not to repeat this again. Thank you for your review. I am incorporating all the changes you suggested. There are parts of the feature which I have not pushed to this branch. hence, you have seen few dangling pieces of code depending on non existing components. Again, I will be fixing them ASAP.  I am converting the current PR to a draft one till I fix everything.

**Comment by wasade** [comment]
Ah that makes sense, thanks @ganeshkumar78

**Pull request closed without merging**